G Suite: Validate the username field when adding new users

klimeryk opened the pull request:
The field with the account/username is named differentely in the two forms we have for setting up G Suite users: in one it's `email`, in the other it's `username`. Since our validation code was only looking for `email`, it did not validate properly the `username` field from the second form and allowed invalid usernames to be submitted.

*Testing*
Try adding new G Suite users and verify that the validation works.

1. Adding to a custom domain you've already purchased on WordPress.com:
 * `Domains` -> `G Suite` -> `Add G Suite`
<img width="763" alt="screen shot 2016-11-01 at 18 52 59" src="https://cloud.githubusercontent.com/assets/3392497/19900658/9a157a98-a064-11e6-886e-306bc225dcaa.png">

 * `Domains` -> select a custom domain -> `Email` -> `Add G Suite`
<img width="762" alt="screen shot 2016-11-01 at 18 54 07" src="https://cloud.githubusercontent.com/assets/3392497/19900675/aa8ea174-a064-11e6-91a0-49a2fd5a8b53.png">

2. Purchasing G Suite along with a new custom domain: `Domains` -> `Add` -> search and select domain -> `Add Email`

<img width="742" alt="screen shot 2016-11-01 at 18 52 08" src="https://cloud.githubusercontent.com/assets/3392497/19900685/b3bde50c-a064-11e6-9478-8acf45f1de3b.png">

Fixes #9053 

cc @Lochlaer 
cc @aidvu @umurkontaci

umurkontaci commented on client/lib/domains/google-apps-users/index.js:
```diff
@@ -37,7 +37,7 @@ function validate( { users, fields, domainSuffix } ) {
 				}
-			} else if ( 'email' === key ) {
+			} else if ( includes( [ 'email', 'username' ], key ) ) {
 				if ( /[^[0-9a-z_'.-]/i.test( field.value ) ) {
```
Quick check: Do we lowercase these inputs somewhere? We should either allow capital letters or lowercase the input if we're not doing so already.

umurkontaci commented:
Codewise this fine, I've left a bunch of comments on where we can do validation but it's not necessary to do them here. But if you choose to do them in a separate PR, can you open a follow up issue?

klimeryk commented:
Thanks for the comments - I'm a sucker for fixing up things like this, so I thought that I'd just fix up the regexp... (which, turns out, is so bad that it allows `[` as a valid character 😅). But, of course, one change led to another :/

Changes:
 * improved regexp - while your comments are valid for email overall, for this use case the requirements set forth by Google in https://support.google.com/a/answer/33386?hl=en are what we should follow. So, no more than one consecutive dot, only certain characters allowed. Apparently, non-alphanumeric characters _are_ ok at the beginning and end of the username. I've opted not to allow the dot at those places, though, for easier implementation, TBH.
 * fixing that ^ uncovered another bug - we were not passing the `domainSuffix` to the validate function... 😓 Since that form can deal with different domains, I had to move the domain part to the user - this is now consistent with the other form.
 * per your comment, username is lowercased and trimmed (good catch!).

I hope this addresses your comments. I'm sure there are other issues (especially with code...), but unless they've been introduced in this PR, let's ignore them for now 😅

aidvu commented:
LGTM, but there is a glitch if I want to add a capital letter in the middle. I also don't like the idea of lowercasing it right away, but that's ok as long as it doesn't move the cursor to the end of the field.

klimeryk commented:
Good point - it _is_ annoying and frustrating when the cursor moves like that. So I'm only trimming the value now (that doesn't move the cursor) and lowercasing it just before it gets added to cart 👍

aidvu commented:
Looking good. There's still a thing I dislike, but it's a very edge case, and I doubt possible. If you click at the beginning and press space, it moves you to the end.

Also pressing space at any point at the edges feels strange (as it trims it). But I guess that was the idea. :) I'd rather trim before passing it to cart, just as you do with lowercase (but then you'd have to change the regexp, right?).

Anyway, I'm fine with it like this.

klimeryk commented:
Heh, that's a nice catch, but indeed a very edgy case ;) I'll merge it as-is, to keep things simple.
Thanks for the review! 🙇
